observations/obs_converters/even_sphere/even_sphere_subdom.m

```diff
@@ -0,0 +1,303 @@
+%
+% The vertical coordinate system is 'pressure'.
+% The pressure levels themselves may be input.
+% The observation error variances for each level must then also be input.
+% There are separate error variances for T and [U,V].
+%
+% Example 1: 30 profiles, 14 default levels and obs error variances
+% nprofiles   = 30;
+% even_sphere_subdom(nprofiles)
+%
+% Example 2: 30 profiles at 6 specified levels, 
+%            uses the default obs error variances for each layer
+% nprofiles   = 30;
+% levels      = [1000  850  500  300  200  100];
+% even_sphere_subdom(nprofiles, 'levels', levels)
+%
+% Example 3: 30 profiles at 6 specified levels, 
+%            specify the error variances for each layer 
+% nprofiles   = 30;
+% levels      = [1000  850  500  300  200  100];
+% T_error_var = [1.44 0.64 0.64 0.81 1.44 0.64];
+% W_error_var = [1.96 2.25 4.41 9.00 7.29 4.41];
+% even_sphere_subdom(nprofiles, 'levels', levels, ...
+%             'T_error_var', T_error_var, 'W_error_var', W_error_var)
+%
+% Example 4: restrict profiles to a horizontal subdomain (which spans the prime meridian).
+% nprofiles   = 2000;  only some of these will be written to the file.
+% lon_left = 300.; left boundary
+% lon_right =  30.; right boundary
+% lat_south = -20.; southern boundary
+% lat_north =  20.; northern boundary
+
+% 
+
+%% DART software - Copyright UCAR. This open source software is provided
+% by UCAR, "as is", without charge, subject to all terms of use at
+% http://www.image.ucar.edu/DAReS/DART/DART_download
+
+%%--------------------------------------------------------------------
+% Decode,Parse,Check the input
+%---------------------------------------------------------------------
+
+p = inputParser;
+
+addRequired(p,'nprofiles',@isnumeric);
+
+% obs types to create at each location.
+obs_types = {'RADIOSONDE_TEMPERATURE','RADIOSONDE_U_WIND_COMPONENT','RADIOSONDE_V_WIND_COMPONENT'};
+num_types = size(obs_types,2);
+
+% These are the *mandatory pressure levels* defined in the
+% AMS glossary https://glossary.ametsoc.org/wiki/Mandatory_level
+% The error variances come from obs_converters/obs_error/ncep_obs_err_mod.f90.
+
+default_levels      = [1000  925  850  700  500  400  300   250  200  150  100   70   50   30   20   10    7    5    3    2    1];
+default_T_error_var = [1.44 1.00 0.64 0.64 0.64 0.64 0.81  1.44 1.44 1.00 0.64 0.64 0.81 1.00 1.69 2.25 2.25 2.25 2.25 2.25 2.25];
+default_W_error_var = [1.96 2.25 2.25 2.56 4.41 6.76 9.00 10.24 7.29 5.76 4.41 4.41 4.41 4.41 4.41 4.41 4.41 4.41 4.41 4.41 4.41];
+default_fill        = false; % For diagnostic test need a null data value and a null qc value
+default_nlevels     = length(default_levels);
+default_YMD         = '2017-12-25';
+default_nlon         = 288;
+default_nlat         = 192;
+default_lon_left         =   0.;
+default_lon_right         = 360.;
+default_lat_south         = -90.;
+default_lat_north         =  90.;
+
+if (exist('inputParser/addParameter','file') == 2)
```

Comment:
it would help if you could add a comment about why one section needs to use addParameter and the other needs to use addParamValue - it's a matlab thing i guess but it's not clear to me what the diff is.

Reply:
I don't know the difference either.  I was just following the template.
I'd need to dig into Matlab to figure it out.  I'll do it if it's important.